kpi/views.py

```diff
@@ -520,7 +525,33 @@ def preview(self, request, *args, **kwargs):
             return Response(response_data, template_name='preview_error.html')
 
 
-class AssetViewSet(viewsets.ModelViewSet):
+class AssetVersionViewSet(NestedViewSetMixin, viewsets.ModelViewSet):
+    model = AssetVersion
+    lookup_field = 'uid'
+    filter_backends = (
+            AssetOwnerFilterBackend,
+        )
+
+    def get_serializer_class(self):
+        if self.action == 'list':
+            return AssetVersionListSerializer
+        else:
+            return AssetVersionSerializer
+
+    def get_queryset(self):
+        _asset_uid = self.get_parents_query_dict()['asset']
+        _deployed = self.request.query_params.get('deployed', None)
+        _queryset = self.model.objects.filter(asset__uid=_asset_uid)
+        if _deployed is not None:
+            _queryset = _queryset.filter(deployed=_deployed)
+        if 'pk' in self.kwargs:
+            return _queryset.filter(uid=self.kwargs['pk'])
```

Comment:
Sorry, the range attached to the comment is misleading. I meant lines 547-8 specifically (now discussing on Flowdock)